src/test/Microsoft.Extensions.DependencyModel.Tests/DependencyContextJsonReaderTest.cs

```diff
@@ -30,7 +30,7 @@ public void ReadsRuntimeTargetInfo()
         ""signature"":""target-signature""
     },
     ""targets"": {
-        "".NETCoreApp,Version=v1.0/osx.10.10-x64"": {},
+        "".NETCoreApp,Version=v1.0/osx.10.10-x64"": {}
```

Comment:
**Question:** Do we expect the json payloads to contain comments? If so, we need to use a more forgiving option within the reader that allows comments.

Reply:
The existing implementation does not support comments inside the payload.

If comments exist, the `DependencyContextJsonReader` throws `FormatException` (since `JsonTextReader` allows comments by default). If the intention was to allow & ignore comments, the `DependencyContextJsonReader` needs to be more robust against the comment token type and ignore it, explicitly.

If we use the new `Utf8JsonReader`, we could either disallow, skip, or allow comments.

1. **[partially breaking change - different exception type]** If we disallow comments (default behavior of the reader), then we throw `JsonReaderException` whenever we see a comment (since it is considered invalid JSON as per spec).
2. **[partially breaking change - what was invalid before is supported now]** If we skip comments, then the behavior is now different (since comments become supported, and ignored).
3. **[non-breaking change]** If we allow comments, then we continue to throw `FormatException`, just like the current implementation.

If we want to match the current behavior (i.e. #3), we have to allow comments to pass through even though that gets detected by `DependencyContextJsonReader` as an error and a `FormatException` is thrown. I much rather have option 1 or 2 where we either support skipping comments (and successfully read the deps file) OR we throw `JsonReaderException` explicitly.

@eerhardt what option would you prefer? I have gone with option 3 in the meantime and added a test encapsulating the current behavior.

Reply:
In this case, I think I'd be fine with changing the exception type, as long as the exception still conveys a decent error message (what is the problem, which line/column, etc). I'd find it hard to believe someone is relying on this behavior, and if they are that they'd have a hard time moving to the new behavior.

Honestly, I'd also be fine with ignoring the comments as well.